File: 03 - Arrays/searching algorithms/linearSearch.py
# implements linear search in python lists (array)

inp = 'hello i am going to implement linear search in an array in python'

array = list(inp.split(" "))
print(array)

key = input("enter a key to be searched in the array : ")

keyFound = False
for index, element in enumerate(array):
    if element == key:
        keyFound = True
        print("Found key ({}) at index {}".format(key, index))
        break

if not keyFound:
    print(f"key ({key}) not found in the array")

File: 03 - Arrays/searching algorithms/linearSearch.c
// implementing linear search in arrays

#include <stdio.h>

void linearSearch(int array[], int size, int key)
{
    int keyFound = 0;
    for(int i = 0; i < size; i++)
    {
        if (array[i] == key)
        {
            keyFound = 1;
            printf("found key (%d) at index %d\n", key, i);
            break;
        }
    }
    if (!keyFound)
    {
        printf("key not found in the array\n");
    }
}

int main()
{
    int numbers[] = {23, 34, 56, 32, 33, 56, 36, 78, 12, 78, 91};
    int size = sizeof(numbers)/sizeof(numbers[0]);

    for (int i = 0; i < size; i++)
    {
        printf("%d ", numbers[i]);
    }
    printf("\n");

    int key;
    printf("enter value to be searched in the given array : ");
    scanf("%d", &key);

    linearSearch(numbers, size, key);

    return 0;
}